maxim/logger/parsers/generation_parser.py

```
@@ -219,8 +220,30 @@ def default_json_serializer(o: Any) -> Any:
+    # by default.
+    if isinstance(o, types.MappingProxyType):
+        return dict(o)
+    # Classes must be handled before the instance-method branches below, since
+    # those methods exist on the class as unbound functions and would raise if
+    # called without an instance. A model *class* is how structured-output
+    # formats are usually declared (e.g. response_format=MyModel), and its
+    # schema is the meaningful representation.
+    if isinstance(o, type):
+        if hasattr(o, "model_json_schema"):  # pydantic v2 model class
+            return o.model_json_schema()
+        if hasattr(o, "schema") and callable(getattr(o, "schema")):  # pydantic v1
```

Ruff (0.15.21)

[warning] Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)